### ISSUE TITLE
repair fails to check parent ids

### ISSUE THREAD

**Issue by niczero** [niczero/minion-backend-storable#8]
Reported by @manuelm in amongst PR #2.

If the job is inactive and has parents, it sets the job to failed saying parents are missing.  It doesn't check whether the parents exists.

**Comment by niczero** [comment]
I think this is a working unit test
```
diff --git a/test/10-file.t b/test/10-file.t
index 1d4f55c..d6d3c26 100644
--- a/test/10-file.t
+++ b/test/10-file.t
@@ -647,6 +647,23 @@ is $minion->job($id)->info->{state},  'failed',           'right state';
 is $minion->job($id)->info->{result}, 'Parent went away', 'right result';
 $worker->unregister;
 
+# Repair orphaned jobs
+$worker = $minion->remove_after(20)->worker->register;
+#$worker = $minion->worker->register;
+$id  = $minion->enqueue('test');
+$id2 = $minion->enqueue('test');
+$id3 = $minion->enqueue(test => [] => {parents => [$id, $id2]});
+is $minion->stats->{delayed_jobs}, 1, 'one delayed job';
+my $quantity_failed = $minion->stats->{failed_jobs};
+$minion->repair;
+is $minion->stats->{delayed_jobs}, 1, 'one delayed job';
+is $minion->stats->{failed_jobs}, $quantity_failed, 'unchanged failed jobs';
+ok $minion->backend->remove_job($id2), 'removed parent job';
+$minion->repair;
+is $minion->stats->{delayed_jobs}, 0, 'no delayed job';
+is $minion->stats->{failed_jobs}, $quantity_failed + 1, 'one more failed job';
+$worker->unregister;
+
 # Worker remote control commands
 $worker  = $minion->worker->register->process_commands;
 $worker2 = $minion->worker->register;
```

**Comment by manuelm** [comment]
Looks sane. Shall I replace mine with yours in my PR?

**Comment by niczero** [comment]
This is fixed in v6.06 e3f9c6d
